### ISSUE TITLE
Lack of Makefile when run the ./build.sh

### ISSUE THREAD

**Issue by HaFred** [ucb-bar/sha3-workload#10]
Hi, I was trying to build the spike-sha3 for the sha3 accelerator with the `build.sh`, I noticed that there is no makefile in the /sha3-workload. Which make file should I try to copy to this file path among its subfolders, would you kindly specify it? @abejgonzalez

`frederick@ubuntu:~/workspace/chipyard/software/firemarshal/workloads/sha3$ ./build.sh
Building Spike
~/workspace/chipyard/software/firemarshal/workloads/sha3/spike-sha3/build ~/workspace/chipyard/software/firemarshal/workloads/sha3
make: *** No targets specified and no makefile found.  Stop.`

Thanks, Abraham.

**Comment by abejgonzalez** [comment]
Hi,

Sorry, this wasn't documented but the `sha3-workload` repo is now deprecated and has moved to be closer to the sources. You should be able to checkout https://github.com/ucb-bar/sha3 and run the `build.sh` within the `software` folder. I'll go ahead and update this repo to reflect those changes (this will be done when we move to the next Chipyard update).

Edit: However, running the `build.sh` should still work in this repository... did you run `git submodule update --init --recursive`

**Comment by HaFred** [comment]
> Hi,
> 
> Sorry, this wasn't documented but the `sha3-workload` repo is now deprecated and has moved to be closer to the sources. You should be able to checkout https://github.com/ucb-bar/sha3 and run the `build.sh` within the `software` folder. I'll go ahead and update this repo to reflect those changes. This will be put in the next Chipyard update.
> 
> Edit: However, running the `build.sh` should still work in this repository... did you run `git submodule update --init --recursive`

Yes, I did run `git submodule update --init --recursive` in my local `sha3-workload`, but it was still not working. I think it makes sense because I notice that there is no `makefile` in the `sha3-workload` repo-master branch, isn't it?

Thanks again for your precious time.

**Comment by HaFred** [comment]
I tried with the https://github.com/ucb-bar/sha3, and run the `build.sh` within `software` folder successfully. However, errors still occured as before:

```
frederick@ubuntu:~/workspace/chipyard/software/firemarshal/workloads/sha3_new/software/benchmarks/src$ riscv64-unknown-elf-gcc sha3-rocc.c 
frederick@ubuntu:~/workspace/chipyard/software/firemarshal/workloads/sha3_new/software/benchmarks/src$ spike --extension=sha3 pk a.out 
couldn't find extension 'sha3' (or library 'libsha3.so')
```

Sorry for the confusion may cause. Initially, I ran into the problem `couldn't find extension 'sha3' (or library 'libsha3.so')`, I thought it could be solved by rebuilding spike for sha3. But even I did so, it still failed. Any suggestions?

Thanks Abraham.

**Comment by abejgonzalez** [comment]
> Yes, I did run `git submodule update --init --recursive` in my local `sha3-workload`, but it was still not working. I think it makes sense because I notice that there is no `makefile` in the `sha3-workload` repo-master branch, isn't it?

Sorry I don't think I understand your question. If you look inside of the `build.sh` script, all it is doing is building a new version of `spike` with the `sha3` extension. It is building it in the `spike-sha3/build` folder. So it will call the `Makefile` that is located in that folder after the `build.sh` configure step. I would double check that you have all sources checked out and delete`spike-sha3/build` if you have it. 

> I tried with the https://github.com/ucb-bar/sha3, and run the `build.sh` within `software` folder successfully. However, errors still occured as before:
> 
> ```
> frederick@ubuntu:~/workspace/chipyard/software/firemarshal/workloads/sha3_new/software/benchmarks/src$ riscv64-unknown-elf-gcc sha3-rocc.c 
> frederick@ubuntu:~/workspace/chipyard/software/firemarshal/workloads/sha3_new/software/benchmarks/src$ spike --extension=sha3 pk a.out 
> couldn't find extension 'sha3' (or library 'libsha3.so')
> ```
> 
> Sorry for the confusion may cause. Initially, I ran into the problem `couldn't find extension 'sha3' (or library 'libsha3.so')`, I thought it could be solved by rebuilding spike for sha3. But even I did so, it still failed. Any suggestions?
> 
> Thanks Abraham.

We need to create a new version of spike located in `spike-local`. So if you call that version of spike directly it should work. Ex:

`./spike-local/bin/spike --extension=sha3 benchmarks/bare/sha3-rocc.riscv`

**Comment by HaFred** [comment]
Thanks Abraham, you are right about the `build.sh`, it works in this way.

One more question, I used spike-sha3 to simulate the `sha3-rocc.riscv` and it took 9 cycles. However, when I tried with `a.out` (`a.out` is the gcc compiled output of `sha3-rocc.c`), it took 11 cycles instead of 9. 

```
frederick@ubuntu:~/workspace/chipyard/software/firemarshal/workloads/sha3_new/software/benchmarks/src$ /home/frederick/workspace/chipyard/software/firemarshal/workloads/sha3_new/isa-sim/spike-local/bin/spike --extension=sha3 /home/frederick/workspace/chipyard/software/firemarshal/workloads/sha3_new/software/benchmarks/bare/sha3-rocc.riscv 
Start basic test 1.
output[0]:203 ==? results[0]:203 
output[1]:52 ==? results[1]:52 
output[2]:27 ==? results[2]:27 
output[3]:85 ==? results[3]:85 
output[4]:46 ==? results[4]:46 
output[5]:79 ==? results[5]:79 
output[6]:152 ==? results[6]:152 
output[7]:228 ==? results[7]:228 
output[8]:86 ==? results[8]:86 
output[9]:138 ==? results[9]:138 
output[10]:201 ==? results[10]:201 
output[11]:206 ==? results[11]:206 
output[12]:253 ==? results[12]:253 
output[13]:168 ==? results[13]:168 
output[14]:255 ==? results[14]:255 
output[15]:107 ==? results[15]:107 
output[16]:122 ==? results[16]:122 
output[17]:177 ==? results[17]:177 
output[18]:65 ==? results[18]:65 
output[19]:68 ==? results[19]:68 
output[20]:231 ==? results[20]:231 
output[21]:19 ==? results[21]:19 
output[22]:70 ==? results[22]:70 
output[23]:198 ==? results[23]:198 
output[24]:64 ==? results[24]:64 
output[25]:90 ==? results[25]:90 
output[26]:192 ==? results[26]:192 
output[27]:80 ==? results[27]:80 
output[28]:206 ==? results[28]:206 
output[29]:234 ==? results[29]:234 
output[30]:168 ==? results[30]:168 
output[31]:159 ==? results[31]:159 
Success!
SHA execution took 9 cycles
frederick@ubuntu:~/workspace/chipyard/software/firemarshal/workloads/sha3_new/software/benchmarks/src$ /home/frederick/workspace/chipyard/software/firemarshal/workloads/sha3_new/isa-sim/spike-local/bin/spike --extension=sha3 pk a.out
bbl loader
Start basic test 1.
output[0]:203 ==? results[0]:203 
output[1]:52 ==? results[1]:52 
output[2]:27 ==? results[2]:27 
output[3]:85 ==? results[3]:85 
output[4]:46 ==? results[4]:46 
output[5]:79 ==? results[5]:79 
output[6]:152 ==? results[6]:152 
output[7]:228 ==? results[7]:228 
output[8]:86 ==? results[8]:86 
output[9]:138 ==? results[9]:138 
output[10]:201 ==? results[10]:201 
output[11]:206 ==? results[11]:206 
output[12]:253 ==? results[12]:253 
output[13]:168 ==? results[13]:168 
output[14]:255 ==? results[14]:255 
output[15]:107 ==? results[15]:107 
output[16]:122 ==? results[16]:122 
output[17]:177 ==? results[17]:177 
output[18]:65 ==? results[18]:65 
output[19]:68 ==? results[19]:68 
output[20]:231 ==? results[20]:231 
output[21]:19 ==? results[21]:19 
output[22]:70 ==? results[22]:70 
output[23]:198 ==? results[23]:198 
output[24]:64 ==? results[24]:64 
output[25]:90 ==? results[25]:90 
output[26]:192 ==? results[26]:192 
output[27]:80 ==? results[27]:80 
output[28]:206 ==? results[28]:206 
output[29]:234 ==? results[29]:234 
output[30]:168 ==? results[30]:168 
output[31]:159 ==? results[31]:159 
Success!
SHA execution took 11 cycles
```

Since the `sha3-rocc.riscv` was built by `sha3-rocc.c`, it is supposed to be the same, right? I wonder do you happen to know why? 

Thanks again.
